projects/packages/jetpack-mu-wpcom/src/class-jetpack-mu-wpcom.php

```diff
@@ -521,4 +522,117 @@ public static function load_social_links() {
+	 * Checks for translation updates and installs them if necessary.
+	 */
+	public static function maybe_update_translations() {
+		$plugin_slug = 'jetpack-mu-wpcom';
+
+		// Get installed site locales.
+		$locales = self::get_all_active_locales();
+		if ( empty( $locales ) ) {
+			return;
+		}
+
+		// Query the translation API.
+		$api_url      = 'https://translate.wordpress.com/api/translations-updates/wpcom/plugins';
+		$request_body = array(
+			'locales' => $locales,
+			'plugins' => array(
+				$plugin_slug => array( 'version' => 'latest' ),
+			),
+		);
+
+		$response = wp_remote_post(
+			$api_url,
+			array(
+				'body'    => wp_json_encode( $request_body ),
+				'headers' => array( 'Content-Type' => 'application/json' ),
+				'timeout' => 10,
+			)
+		);
+
+		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
+			return;
+		}
+
+		$data = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		if ( empty( $data['data'][ $plugin_slug ] ) ) {
+			return;
+		}
+
+		// Process each locale.
+		foreach ( $data['data'][ $plugin_slug ] as $translation ) {
+			$package_url   = $translation['package'] ?? '';
+			$locale        = $translation['wp_locale'] ?? '';
+			$last_modified = $translation['last_modified'] ?? '';
+
+			if ( ! $package_url || ! $locale || ! $last_modified ) {
+				continue;
+			}
+
+			// Get last saved timestamp for this locale.
+			$stored_last_modified = get_option( "jetpack_mu_wpcom_translation_{$locale}_last_modified", '' );
+
+			// If translation is already the latest version, skip.
+			if ( $stored_last_modified === $last_modified ) {
+				continue;
+			}
+
+			// Install the translation package.
+			if ( self::install_translation_package( $package_url ) ) {
+				update_option( "jetpack_mu_wpcom_translation_{$locale}_last_modified", $last_modified );
+			}
+		}
+	}
+
+	/**
+	 * Retrieves all active site locales.
+	 */
+	private static function get_all_active_locales() {
+		$locales             = array( get_locale() );
+		$available_languages = get_available_languages();
+		if ( ! empty( $available_languages ) ) {
+			$locales = array_unique( array_merge( $locales, $available_languages ) );
+		}
+		return array_values( $locales );
+	}
+
+	/**
+	 * Downloads and installs translation files from the given URL.
+	 *
+	 * @param string $package_url The URL of the language pack.
+	 *
+	 * @return bool True on success, false on failure.
+	 */
+	private static function install_translation_package( $package_url ) {
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+
+		$destination = WP_LANG_DIR . '/mu-plugins/';
+
+		// Ensure the mu-plugin's languages directory exists.
+		if ( ! is_dir( $destination ) ) {
+			wp_mkdir_p( $destination );
+		}
+
+		$temp_file = download_url( $package_url );
+		if ( is_wp_error( $temp_file ) ) {
+			return false;
+		}
+
+		// Extract language files into the destination directory.
+		$unzip_result = unzip_file( $temp_file, $destination );
```

Comment:
Do we need an equivalent of core's `Language_Pack_Upgrader->clear_destination()` before we unzip the file?

Or I wonder if we could reuse `Language_Pack_Upgrader` to do most of the work for us by passing it the right information. At a quick glance, maybe, although we may have to use the 'upgrader_package_options' hook to override the destination to `WP_LANG_DIR . '/mu-plugins/'` if we want that instead of `/plugins`.

Reply:
That’s a great suggestion! Initially, I was worried that [clear_destination](https://developer.wordpress.org/reference/classes/language_pack_upgrader/clear_destination/) might delete all translation files inside `wp-content/languages/mu-plugins/`, potentially affecting other MU plugins like `wpcomsh` (when we migrate those files to this directory). However, after reviewing the actual implementation, I confirmed that it only deletes translation files specifically for the language and plugin being updated.

Reply:
Adding `Language_Pack_Upgrader` adds unexpected dependencies, and a lot of warning messages regarding [undefined](https://github.com/WordPress/wordpress-develop/blob/8a52d746e9bb85604f6a309a87d22296ce1c4280/src/wp-admin/includes/class-language-pack-upgrader.php#L413) [properties](https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-admin/includes/class-language-pack-upgrader.php#L416). I'll implement it manually to reduce reliance on these dependencies.